capa/features/extractors/dnfile/helpers.py

```diff
@@ -299,22 +316,93 @@ def get_dotnet_unmanaged_imports(pe: dnfile.dnPE) -> Iterator[DnUnmanagedMethod]
         # like kernel32.CreateFileA
         yield DnUnmanagedMethod(token, module, method)
 
+def get_nested_class_table(pe):
+    nested_class_table = {}
+    
+    # Used to find nested classes in typedef
+    for rid, nestedclass in iter_dotnet_table(pe, dnfile.mdtable.NestedClass.number):
+        assert isinstance(nestedclass, dnfile.mdtable.NestedClassRow)
+        nested_class_table[nestedclass.NestedClass.row_index] = nestedclass.EnclosingClass.row_index
+
+    return nested_class_table
 
+def get_typedef_class_table(pe):
+    typedef_class_table = []
+    
+    for rid, typedef in iter_dotnet_table(pe, dnfile.mdtable.TypeDef.number):
+        assert isinstance(typedef, dnfile.mdtable.TypeDefRow)
+        typedef_class_table.append((typedef.TypeName, typedef.TypeNamespace))
+        
+    return typedef_class_table
+    
+def typedef_helper(index, nested_class_table, typedef_class_table, n, name):
+    # Append the current typeref name
+    n.append(name)
+    
+    if nested_class_table[index] in nested_class_table:
+        typedef_helper(nested_class_table[index], nested_class_table, typedef_class_table, n, typedef_class_table[nested_class_table[index]-1][0])
+    else:
+        # Document the root enclosing details
+        n.append(typedef_class_table[nested_class_table[index]-1][0])
+        namespace = typedef_class_table[nested_class_table[index]-1][1]
+        n.append(namespace)
+        
+    return tuple(n[::-1])
+
+def get_typeref_table(pe):
+    typeref_table = []
+    
+    # Used to track values in typeref table
+    for rid, typeref in iter_dotnet_table(pe, dnfile.mdtable.TypeRef.number):
+        assert isinstance(typeref, dnfile.mdtable.TypeRefRow)
+        typeref_table.append((typeref.TypeName, typeref.TypeNamespace, typeref.ResolutionScope.row_index, type(typeref.ResolutionScope.table)))
+        
+    return typeref_table
+
+def typeref_helper(index, typeref_table, n, name):
+    # Append the current typeref name
+    n.append(name)
+    
+    if typeref_table[index - 1][3] == dnfile.mdtable.TypeRef:
+        # Recursively call helper function with enclosing typeref details
+        typeref_helper(typeref_table[index - 1][2], typeref_table, n, typeref_table[index - 1][0])
+    else:
+        # Document the root enclosing details
+        n.append(typeref_table[index - 1][0])
+        namespace = typeref_table[index - 1][1]
+        n.append(namespace)
+        
+    return tuple(n[::-1])
```

Comment:
Pending your thoughts on my comment about making this function iterative, consider returning `namespace, tuple(class1, class2, ...)` here so it's more intuitive for the caller to unpack.